Gems/ROS2/Code/Source/RobotImporter/URDF/UrdfParser.cpp

```diff
@@ -93,13 +94,17 @@ namespace ROS2::UrdfParser
         ParseResult parseResult;
         std::ostringstream parseStringStream;
         {
-            RedirectSDFOutputStream redirectConsoleStream(sdf::Console::Instance()->GetMsgStream(), parseStringStream);
+            RedirectSDFOutputStream redirectConsoleStreamMsg(sdf::Console::Instance()->GetMsgStream(), parseStringStream);
             parseResult.m_sdfErrors = parseResult.m_root.LoadSdfString(xmlString, parserConfig);
             // Get any captured sdf::Error messages
         }
+        const auto & parseMessages = parseStringStream.str();
 
-        std::string parseMessages = parseStringStream.str();
-        parseResult.m_parseMessages = AZStd::string(parseMessages.c_str(), parseMessages.size());
+        // regular exprsion to escape console's color codes
+        const AZStd::regex escapeColor("\x1B\\[[0-9;]*[A-Za-z]");
+
+        parseResult.m_parseMessages =  AZStd::regex_replace(AZStd::string(parseMessages.c_str(), parseMessages.size()),escapeColor, "");
+        AZ_Printf("ROS2", "SDF Stream: %s", parseResult.m_parseMessages.c_str());
```

Comment:
There is no need to print the messages here as it gets printed in the added logic in the RobotImporterWidget.cpp OpenUrdf function.

The entire purpose of returning the parse messages as part of a result structure is to allow the **caller** to decide to print out the message, not function implementer.

Reply:
Ok - I kept it to have this output present when URDF is loaded with ImportBus.
I've opened an issue to adjust the robot importer bus to follow your recommendations https://github.com/o3de/o3de-extras/issues/500.